application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml

```diff
@@ -172,13 +180,14 @@
     }
     // Get the params.
     var params = window.flashNotifierParams[window.flashNotifierIndex];
-    // Show the popup.
-    $(window).on('load', function() {
-      window.flashPopup.modal('show');
-    });
     // When window has already been loaded.
-    if(window.flashNotifierIndex &gt; 0) {
+    if(document.readyState === 'complete') {
       window.flashPopup.modal('show');
+    } else {
+      // Show the popup.
+      $(window).on('load', function() {
+        window.flashPopup.modal('show');
+      });
```

Comment:
I think you can replace all this with:

```
$($.proxy(window.flashPopup, 'modal', 'show'));
```

See https://api.jquery.com/ready/#ready-handler